src/algorithms/tracking/CKFTracking.cc

```diff
@@ -123,24 +159,92 @@ namespace eicrecon {
                 m_geoctx, m_fieldctx, m_calibctx, slAccessorDelegate,
                 extensions, Acts::LoggerWrapper{logger()}, pOptions, &(*pSurface));
 
-        auto results = (*m_trackFinderFunc)(init_trk_params, options);
+        auto results = (*m_trackFinderFunc)(acts_init_trk_params, options);
 
-        for (std::size_t iseed = 0; iseed < init_trk_params.size(); ++iseed) {
+        for (std::size_t iseed = 0; iseed < acts_init_trk_params.size(); ++iseed) {
 
             auto &result = results[iseed];
 
             if (result.ok()) {
+
                 // Get the track finding output object
                 auto &trackFindingOutput = result.value();
+
                 // Create a SimMultiTrajectory
-                trajectories.push_back(new ActsExamples::Trajectories(std::move(trackFindingOutput.fittedStates),
-                                                                              std::move(trackFindingOutput.lastMeasurementIndices),
-                                                                              std::move(trackFindingOutput.fittedParameters)));
+                auto* multiTrajectory = new ActsExamples::Trajectories(
+                    std::move(trackFindingOutput.fittedStates),
+                    std::move(trackFindingOutput.lastMeasurementIndices),
+                    std::move(trackFindingOutput.fittedParameters)
+                );
+
+                // Get the entry index for the single trajectory
+                // The trajectory entry indices and the multiTrajectory
+                const auto& mj        = multiTrajectory->multiTrajectory();
+                const auto& trackTips = multiTrajectory->tips();
+                if (trackTips.empty()) {
+                    m_log->debug("Empty multiTrajectory.");
+                    continue;
+                }
+
+                const auto& trackTip = trackTips.front();
+
+                // Collect the trajectory summary info
+                auto trajectoryState = Acts::MultiTrajectoryHelpers::trajectoryState(mj, trackTip);
+
+                // Create trajectory
+                auto trajectory = trajectories->create();
+
+                trajectory.setChi2(trajectoryState.chi2Sum / trajectoryState.NDF);
+                trajectory.setNdf(trajectoryState.NDF);
+
```

Comment:
We should add 

`trajectory.nMeasurements(trajectoryState.nMeasurements);`

etc...

Reply:
Based on the description here ( https://github.com/eic/EDM4eic/blob/main/edm4eic.yaml#L337), we should do
`
trajectory.setChi2(trajectoryState.chi2Sum);`

Reply:
Yeah, I was following the EDM4hep plugin convention and didn't know we had picked the opposite convention already.